### DISCUSSION TITLE
Which is better board ESP32 MH-ET LIVE mini or ESP32-S2 lolin mini?

### DISCUSSION THREAD

**Discussion by cheolho-kang** [awawa-dev/HyperSerialESP32#20] [Q&A]
Hello, I found HyperSerialESP8266 while looking for a way to control 220 LEDs for a TV backlight.
This seems a awesome solution!
I recently had a problem with the WLED due to data transfer speed issues and your software solved it. I'm really thank you.

I'm looking into a new board because my board(**ESP8266 CP2102 board**) can support only 1M baudrate.
However, looking at your latest v8.0  update log, I can't figure out which one is the best.

**Performance result (version: 8.0, serial port speed: 2M)**
---
RGBW LED strip / Device | ESP32MH-ET LIVE mini|ESP32-S2 lolin mini
-- | -- | --
300LEDsRefresh rate/continues output=100Hz| 93-97|80-84

=> As a result, ESP32MH-ET LIVE mini looks like better.

**Capable Serial Port Speed**
---
ESP32MH-ET LIVE mini|ESP32-S2 lolin mini
| --|--
 4M | 5M

=> But considering the maximum serial port speed it supports, ESP32-S2 lolin mini looks better.

Could you recommend which board better to buy?

And also as you mentioned other comments, it seems that I can speed up to 20Mb through HyperSPI with my current board (Node MCU ESP8266 CP2102).
https://github.com/awawa-dev/HyperSerialEsp8266/discussions/5#discussioncomment-1255941

So I'm assuming it would be better than buying another new board.

Here is my summarized questions:
- I would like to ask your opinion
1.  Is it better to use HyperSPI with my board?
2. Do you think using HyperESP32 is better than HyperESP8266 with restricted 1Mb baudrate?
2-1. Which board do you think would be better to buy?

**Comment by awawa-dev** [comment]
Hi
1. To clarify one thing in your table: ESPSP32MH-ET LIVE mini reaches almost 100 FPS in multi-segment mode (one LED strip divided into 2 equal parts, you must build your setup this way first or divide existing which may be difficult to put the second data-line in the middle of your current LED strip). 
In single segment mode they have almost the same results. Currently in v8 ESP32-S2 is not recommended for multi-segment setup but incoming PR for parallel multi-segment in v9 changes that thing also: both boards performs almost the same.I have been testing the ESP32-S2 intensively for 3 months and it performs well on my system (Rpi OS, Windows) but have limited feedback from users yet (seems even webos Linux also supports it): it's a fresh solution.
2. ` Which board do you think would be better to buy?` HyperSerialEsp8266 has about 10% less performance than HyperSerialESP32, 1Mb doesn't improves the thing also. Definitely ESP32 is a better solution. It can be almost as cheap as ESP8266 and for ESP8266 there are a lot of counterfeited CH340 chips which don't support 2Mb speed.
3. `Is it better to use HyperSPI with my board?` For your current setup it could be an alternative if you dont want to buy new ESP32 board.

**Reply by cheolho-kang** [reply]
Oh. Thank you for your quickly reply.
Here is my setup environment

|   |  |
|---|---|
| Raspberry board | RaspberryPi4 (with several docker) |
| ESP board | ESP8266 CP2102 board |
| LED Strip | SK6812 Natural White 221 LEDs |
| Power Supply | 5V 12A |
| Video Capture | https://a.aliexpress.com/_mO3KQRU |


![roadmap](https://postfiles.pstatic.net/MjAyMzAxMTFfMjU3/MDAxNjczNDQwMzkzNjE2.7QUxjkJDay6cSmv2AAId5K_fCAQ8YQUMh4lErMFEbEAg.bgeMVQsqYC-9Pev0ixEdYrC02bGs3eEg9oRM85nLo_cg.PNG.kbright0912/image.png?type=w966)
-> Changed from WLED to HyperSerialESP8266

1. As you said, performance is similar for single-segment. Then I think ESP32-S2 might be better in my setup (single segment).  Because ESP32-S2 can serial port speed up to 5Mb.

2. How about the comparing HyperSPI and HyperSerialESP32, which do you think is the better solution? (I don't care about buying new board)

**HyperSPI**
Pros: Can speed up to 20Mb
Cons: I think that makes more overhead to on RaspberryPi4 on multi tasking environment. (not sure)

**HyperSerialESP32**
I'm curious if you recommend that it's better than HyperSPI even though it can only use port speed until 5Mb.

**Reply by awawa-dev** [reply]
For single segment speed higher than 2Mb means only lower latency (you could calculate how long does it takes to send a package using 2Mb or 4Mb, the packet size is aprox. 4bytes * RGBW LED number). 1Mb may not be able to provide stable transmission for larger setups and to saturate LED Neopixel bus at 800kbp due to serial communication overhead and processing, but 2Mb should be rather enough. Multi-segment is completely different subject and 4Mb should be used at least. Or higher for lower latency.

>HyperSPI
Pros: Can speed up to 20Mb
Cons: I think that makes more overhead to on RaspberryPi4 on multi tasking environment. (not sure)

SPI is very light transmission and for sure it's lighter than serial communication than must be handled by the kernel driver.

> HyperSerialESP32
I'm curious if you recommend that it's better than HyperSPI even though it can only use port speed until 5Mb

I explained how the speed affects the final effect at the beginning. HyperSPI currently doesn't support multi-segment but it may change in the future.

**Comment by kenny-nt** [comment]
> For single segment speed higher than 2Mb means only lower latency (you could calculate how long does it takes to send a package using 2Mb or 4Mb, the packet size is aprox. 4bytes * RGBW LED number). 1Mb may not be able to provide stable transmission for larger setups and to saturate LED Neopixel bus at 800kbp due to serial communication overhead and processing, but 2Mb should be rather enough. Multi-segment is completely different subject and 4Mb should be used at least. Or higher for lower latency.
> 
> > HyperSPI
> > Pros: Can speed up to 20Mb
> > Cons: I think that makes more overhead to on RaspberryPi4 on multi tasking environment. (not sure)
> 
> SPI is very light transmission and for sure it's lighter than serial communication than must be handled by the kernel driver.
> 
> > HyperSerialESP32
> > I'm curious if you recommend that it's better than HyperSPI even though it can only use port speed until 5Mb
> 
> I explained how the speed affects the final effect at the beginning. HyperSPI currently doesn't support multi-segment but it may change in the future.

If 300 leds > HyperSPI can handle single segment easily?
